nfc/NfcCx/Cx/device.cpp

```diff
@@ -1022,20 +1028,44 @@ Return Value:
 
     TRACE_FUNCTION_ENTRY(LEVEL_VERBOSE);
 
-    TRACE_LINE(LEVEL_INFO,
-               "Queue 0x%p, Request 0x%p, OutputBufferLength %Iu, InputBufferLength %Iu, IoControlCode %!NFC_IOCTL!",
-               Queue,
-               Request,
-               OutputBufferLength,
-               InputBufferLength,
-               IoControlCode);
-
     fdoContext = NfcCxFdoGetContext(WdfIoQueueGetDevice(Queue));
     fileContext = NfcCxFileGetContext(WdfRequestGetFileObject(Request));
 
     for (i = 0; i < ARRAYSIZE(g_NfcCxDdiModules); i++) {
 
-        if (g_NfcCxDdiModules[i].IsIoctlSupported(fdoContext, IoControlCode)) {
+        if (fileContext->Role == ROLE_EMBEDDED_SE) {
+            if (g_NfcCxDdiModules[5].IsIoctlSupported(fdoContext, IoControlCode)) {
+
+                if (NULL == fileContext &&
+                    FALSE == g_NfcCxDdiModules[5].IsNullFileObjectOk) {
+                    TRACE_LINE(LEVEL_ERROR, "%S module request received without a file context", g_NfcCxDdiModules[5].Name);
+                    status = STATUS_INVALID_DEVICE_STATE;
+                    goto Done;
+                }
+
+                if (NULL != fileContext &&
+                    fileContext->IsAppContainerProcess &&
+                    FALSE == g_NfcCxDdiModules[5].IsAppContainerAllowed) {
+                    TRACE_LINE(LEVEL_ERROR, "%S module request received received from the AppContainer process", g_NfcCxDdiModules[5].Name);
+                    status = STATUS_INVALID_DEVICE_STATE;
+                    goto Done;
+                }
+
+                status = g_NfcCxDdiModules[5].IoDispatch(fileContext,
+                    Request,
+                    OutputBufferLength,
+                    InputBufferLength,
+                    IoControlCode);
+                if (!NT_SUCCESS(status)) {
+                    TRACE_LINE(LEVEL_ERROR, "Failed to forward the request to the %S module, %!STATUS!", g_NfcCxDdiModules[5].Name, status);
+                    goto Done;
+                }
+
+                break;
+            }
+        }
+        else {
```

Comment:
this comment is addressed and the code will be changed, if there is a change in design